Source/Tools/LinuxEmulation/LinuxSyscalls/FileManagement.h

```diff
@@ -88,7 +88,60 @@ class FileManager final {
 
   bool SupportsProcFSInterpreterPath() const { return SupportsProcFSInterpreter; }
 
+#if defined(ASSERTIONS_ENABLED) && ASSERTIONS_ENABLED
+  void TrackFEXFD(int FD) noexcept {
+    std::lock_guard lk(FEXTrackingFDMutex);
+    FEXTrackingFDs.emplace(FD);
+  }
+
+  void RemoveFEXFD(int FD) noexcept {
+    std::lock_guard lk(FEXTrackingFDMutex);
+    FEXTrackingFDs.erase(FD);
+  }
+
+  void RemoveFEXFDRange(int begin, int end) noexcept {
+    std::lock_guard lk(FEXTrackingFDMutex);
+
+    // Just linear scan since the number of tracking FDs is low.
+    for (auto it = FEXTrackingFDs.begin(); it != FEXTrackingFDs.end(); ) {
```

Comment:
Seems like this can be:

```cpp
std::erase_if(FEXTrackingFDs, [begin, end](int FD) {
  return FD >= begin && (FD <= end || end == -1);
});
```

if I'm reading this right

Reply:
You're right, changed.